### PR TITLE
concurix: use node IPC from tracer to pm for traces

### PR DESCRIPTION
**Pull request by Setogit**
Connect to strongloop-internal/scrum-nodeops#330
Connect to strongloop-internal/scrum-nodeops#332

### CODE REVIEW

**Comment by Setogit** [comment]
all comments incorporated to PRs.  Confirmed the data path from tracer to superviso  mesh server to minkelite DB works.

**Comment by kraman** [comment]
@slnode test please

**Comment by rmg** [comment]
CI failed because minkelite was not on CI and therefore not being published to staging.

**Comment by rmg** [comment]
@slnode test please

**Comment by kraman** [comment]
Add options argument to server which allows changes to configuration of Minkelite.

`trace.enableDebugServer`: Boolean. Turn on minkelite server
`trace.debugServerPort`: Number. Debug server port. Defaults to 8104
`trace.db.inMemory`: Boolean. Store data in memory
`trace.db.name`: String. Name of trace database. Default to minkelite.db
`trace.db.path`: String. Path to store database. (This is required)
`trace.data.chartMinutes`: Number. Minutes of trace data to keep. Defaults to 1440 minutes.
`trace.data.staleMinutes`: Number. Minutes of data to keep. Defaults to 1450 minutes.
`trace.data.maxTransaction`: Number. Number of transactions to keep. Default to 100

**Comment by kraman** [comment]
@Setogit @sam-github @seanbrookes PTAL

**Comment by kraman** [comment]
@Setogit Please note that `trace.db.path` must be provided by strong-pm. Please update your strong-pm PR to match

**Comment by sam-github** [comment]
LGTM, if it works for @seanbrookes

**Comment by kraman** [comment]
@sam-github. The story to enable @seanbrookes is a separate POC and requires PR in loopback and loopback-boot to complete.

**Comment by Setogit** [comment]
trace.data.maxTransaction: Number.  -- just to clarify, it specifies number of transactions returned by the getTransation API (JS or HTTP).   We store all transactions to MinkeLite DB.  This number impacts the overall API performance a lot.  I recommend default to be 30.  In general, we should use the same defaults as used in the Concurix version.  Likewise, 8103 is recommended for default of trace.debugServerPort.  It's arbitrary number, just for a historical reason.

So, in summary, all recommended defaults:
    trace.enableDebugServer: FALSE
    trace.debugServerPort:  8103
    trace.db.inMemory: FALSE
    trace.db.name: minkelite.db
    trace.db.path: <see my comment below>
    trace.data.chartMinutes:  1440, i.e., 24 hours
    trace.data.staleMinutes: 1450, chartMinutes + 10 is recommended
    trace.data.maxTransaction:30


RE: trace.db.path -- @sam, @kraman, and @seto agreed we can use the existing this._baseDir.

**Comment by kraman** [comment]
@Setogit PTAL and verify information in 4a299b0. I added some docs explaining the options.

**Comment by kraman** [comment]
@slnode test please

**Comment by kraman** [comment]
* Rebased onto master
 * Tracing is disabled by default unless `trace.enable` option must be set

**Comment by kraman** [comment]
@Setogit ... strong-pm tracing PR will need to be modified accordingly to enable tracing if `--trace` option is passed in.

/cc @sam-github

**Comment by kraman** [comment]
@slnode test please

**Comment by Setogit** [comment]
@kraman -- Since all the Trace changes were incorporated to mesh model, effectively here is no such thing as "strong-pm tracing PR" anymore if I understand it correctly.  So, NO PRs planned to strong-pm for #Trace milestone from my side.  Did I miss anything?

**Comment by kraman** [comment]
@Setogit No, A strong-pm PR is still needed for 2 things:
 * strong-pm should have a `--trace` option to enable tracing and set `options['trace.enable'] = true;`
 * strong-pm should set `options['trace.db.path']`

**Comment by kraman** [comment]
Depends on Depends on https://github.com/strongloop/loopback-boot/pull/106

**Comment by kraman** [comment]
@slnode test please

**Comment by sam-github** [comment]
LGTM
